src/components/TodoList/TodoList.tsx

```diff
@@ -1,100 +1,118 @@
-        <td className="is-vcentered" />
-        <td className="is-vcentered is-expanded">
-          <p className="has-text-danger">quis ut nam facilis et officia qui</p>
-        </td>
-        <td className="has-text-right is-vcentered">
-          <button data-cy="selectButton" className="button" type="button">
-            <span className="icon">
-              <i className="far fa-eye-slash" />
-            </span>
-          </button>
-        </td>
-      </tr>
+import { TodoContext } from '../../Contexts/TodoContext';
 
-      <tr data-cy="todo" className="">
-        <td className="is-vcentered">1</td>
-        <td className="is-vcentered" />
-        <td className="is-vcentered is-expanded">
-          <p className="has-text-danger">delectus aut autem</p>
-        </td>
-        <td className="has-text-right is-vcentered">
-          <button data-cy="selectButton" className="button" type="button">
-            <span className="icon">
-              <i className="far fa-eye" />
-            </span>
-          </button>
-        </td>
-      </tr>
+export const TodoList = () => {
+  const [loader, setLoader] = useState(true);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const { filterBy } = useContext(TodoContext);
+  const { searchQuery } = useContext(TodoContext);
+  const { currentTodo, setCurrentTodo } = useContext(TodoContext);
 
-      <tr data-cy="todo" className="">
-        <td className="is-vcentered">6</td>
-        <td className="is-vcentered" />
-        <td className="is-vcentered is-expanded">
-          <p className="has-text-danger">
-            qui ullam ratione quibusdam voluptatem quia omnis
-          </p>
-        </td>
-        <td className="has-text-right is-vcentered">
-          <button data-cy="selectButton" className="button" type="button">
-            <span className="icon">
-              <i className="far fa-eye" />
-            </span>
-          </button>
-        </td>
-      </tr>
+  const filteredTodos = todos.filter(todo => {
+    if (filterBy !== 'all') {
+      const isCompleted =
+        filterBy === 'completed' ? todo.completed : !todo.completed;
 
-      <tr data-cy="todo" className="">
-        <td className="is-vcentered">8</td>
-        <td className="is-vcentered">
-          <span className="icon" data-cy="iconCompleted">
-            <i className="fas fa-check" />
-          </span>
-        </td>
-        <td className="is-vcentered is-expanded">
-          <p className="has-text-success">quo adipisci enim quam ut ab</p>
-        </td>
-        <td className="has-text-right is-vcentered">
-          <button data-cy="selectButton" className="button" type="button">
+      if (!isCompleted) {
+        return false;
+      }
+    }
+
+    if (
+      searchQuery &&
+      !todo.title.toLowerCase().includes(searchQuery.toLowerCase())
+    ) {
+      return false;
+    }
+
+    return true;
+  });
+
+  {
+    /* eslint-disable react-hooks/exhaustive-deps */
+  }
```

Comment:
most often this means you are doing something wrong, as this lint rule is here for the reason